addons/purchase_stock/models/stock_move.py

```diff
@@ -217,7 +217,8 @@ def _get_related_invoices(self):
         """ Overridden to return the vendor bills related to this stock move.
         """
         rslt = super(StockMove, self)._get_related_invoices()
-        rslt += self.mapped('picking_id.purchase_id.invoice_ids').filtered(lambda x: x.state == 'posted')
+        purchase_ids = self.env['purchase.order'].search([('picking_ids', 'in', self.picking_id._ids)])
```

Comment:
if I'm not mistaken, `_ids` contains `NewIds`, which don't exists in database, so it's useless to look for them. Use only `ids` instead.
Also what is the `len(ids)`? Because a query with a gigantic list of ids is another problematic use-case. :S

Reply:
This has been rewritten in the new fix